docs/api-diffs/9.0/rc.1/diff.md

```diff
@@ -0,0 +1,1290 @@
++        public interface ICloudFormationTemplateResource : IAWSResource, ICloudFormationResource, IResource
+     }
++    namespace Aspire.Hosting.AWS.Provisioning {
++        public class AWSProvisioningException : Exception {
++            public AWSProvisioningException(string message, Exception? innerException = null);
++        }
++    }
+ }
+```
+
+## Aspire.Hosting.Azure
+
+``` diff
+ {
+     namespace Aspire.Hosting {
+         public static class AzureBicepResourceExtensions {
+-            public static IResourceBuilder<AzureBicepResource> AddBicepTemplate(this IDistributedApplicationBuilder builder, string name, string bicepFile);
++            public static IResourceBuilder<AzureBicepResource> AddBicepTemplate(this IDistributedApplicationBuilder builder, [ResourceNameAttribute] string name, string bicepFile);
+-            public static IResourceBuilder<AzureBicepResource> AddBicepTemplateString(this IDistributedApplicationBuilder builder, string name, string bicepContent);
++            public static IResourceBuilder<AzureBicepResource> AddBicepTemplateString(this IDistributedApplicationBuilder builder, [ResourceNameAttribute] string name, string bicepContent);
++            public static IResourceBuilder<T> WithParameter<T>(this IResourceBuilder<T> builder, string name, EndpointReference value) where T : AzureBicepResource;
++            public static IResourceBuilder<T> WithParameter<T>(this IResourceBuilder<T> builder, string name, ReferenceExpression value) where T : AzureBicepResource;
+         }
+         public class AzureConstructResource : AzureBicepResource {
++            public ProvisioningContext? ProvisioningContext { get; set; }
+         }
+         public static class AzureConstructResourceExtensions {
+-            public static IResourceBuilder<AzureConstructResource> AddAzureConstruct(this IDistributedApplicationBuilder builder, string name, Action<ResourceModuleConstruct> configureConstruct);
++            public static IResourceBuilder<AzureConstructResource> AddAzureConstruct(this IDistributedApplicationBuilder builder, [ResourceNameAttribute] string name, Action<ResourceModuleConstruct> configureConstruct);
++            public static ProvisioningParameter AsProvisioningParameter(this IResourceBuilder<ParameterResource> parameterResourceBuilder, ResourceModuleConstruct construct, string? parameterName = null);
++            public static ProvisioningParameter AsProvisioningParameter(this BicepOutputReference outputReference, ResourceModuleConstruct construct, string? parameterName = null);
+-            public static void AssignProperty<T>(this Resource<T> resource, Expression<Func<T, object?>> propertySelector, IResourceBuilder<ParameterResource> parameterResourceBuilder, string? parameterName = null);
+-            public static void AssignProperty<T>(this Resource<T> resource, Expression<Func<T, object?>> propertySelector, BicepOutputReference outputReference, string? parameterName = null);
```

Comment:
Breaking change

Reply:
With what API to replace those? Are there examples?